ui/signed-in/src/commonMain/kotlin/social/androiddev/signedin/composables/SignedInRootContent.kt

```diff
@@ -38,12 +49,43 @@ fun SignedInRootContent(
     modifier: Modifier = Modifier,
 ) {
     val childStack by component.childStack.subscribeAsState()
+    var headerSettings by remember { mutableStateOf<NavHeaderSettings>(NavHeaderSettings.Empty) }
 
-    // TODO Add bottom navigation
-    Box(
+    LaunchedEffect(childStack.active) {
+        headerSettings = when (childStack.active.instance) {
+            is SignedInRootComponent.Child.Settings -> {
+                NavHeaderSettings.Settings
+            }
+
+            is SignedInRootComponent.Child.Timeline -> {
+                NavHeaderSettings.Timeline(action = component::navigateToSettings)
+            }
+        }
+    }
+    Column(
         modifier = modifier,
-        contentAlignment = Alignment.Center
     ) {
+        TopAppBar(
+            title = { Text(headerSettings.title) },
+            actions = {
+                headerSettings.headerAction?.let { header ->
+                    IconButton(onClick = header.action) {
+                        Icon(header.icon, null)
+                    }
+                }
+            },
+            navigationIcon = if (childStack.backStack.isNotEmpty()) {
+                (
+                    {
+                        IconButton(onClick = component::navigateBack) {
+                            Icon(Icons.Default.ArrowBack, null)
+                        }
+                    }
+                    )
+            } else {
+                null
+            }
+        )
```

Comment:
here's a handy util funtion to add to the project
```kotlin
/**
 * Only return the result of [block] if [expression] is true, otherwise always returns null
 */
inline fun <T> nullUnless(expression: Boolean, block: () -> T): T? {
    contract {
        returnsNotNull() implies (expression)
        callsInPlace(block, InvocationKind.AT_MOST_ONCE)
    }
    return if (!expression) {
        null
    } else {
        block()
    }
}
```

and then this could be more readable and simplified to:
```kotlin
navigationIcon = nullUnless(childStack.backStack.isNotEmpty()) {
   {
         IconButton(onClick = component::navigateBack) {
               Icon(Icons.Default.ArrowBack, null)
         }
   }
}
```

Reply:
Way better.
Any suggestion in which module the function should live? We have some util in `network`, but I imagine the function could be used in most places

Reply:
Probably we should create a `kotlin-util` module, and any/all modules can depend on it if they want to. I have some more kotlin util functions I use that I will probably add. 

(Also runCatchingIgnoreCancelled` is one I already added to this project and should/could probably move to this new kotlin-util module as well